src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectivesSpec.scala

```diff
@@ -105,6 +105,53 @@ class StandardUserInfoDirectivesSpec extends FlatSpec with PropertyBasedTesting
     }
   }
 
+  val validJwt = "eyJhbGciOiJSUzI1NiIsImtpZCI6InB1YmxpYzozZWQxZTgwYy1kZTBkLTQwNTctYTQwZi05ZDgxZjRlMzM2OTQiLCJ0eXAiOiJKV1QifQ.eyJhdWQiOlsiOTAzY2ZhZWItNTdkOS00ZWY2LTU2NTktMDQzNzc3OTRlZDY1Il0sImNsaWVudF9pZCI6IjkwM2NmYWViLTU3ZDktNGVmNi01NjU5LTA0Mzc3Nzk0ZWQ2NSIsImV4cCI6MTU4MTYwNTIzMCwiZXh0Ijp7ImlkZW50aXRpZXMiOlsiZHZvZXR0ZXN0QGdtYWlsLmNvbSJdfSwiaWF0IjoxNTgxNjAxNjI5LCJpc3MiOiJodHRwczovL2ljLWRvdC1kb2dlaWN0ZXN0LmFwcHNwb3QuY29tLyIsImp0aSI6Ijc0ZGFkNTFkLTEwZTAtNDk4OC1iOGRlLWI2YTkzYmExODcxMCIsIm5iZiI6MTU4MTYwMTYyOSwic2NwIjpbIm9wZW5pZCIsIm9mZmxpbmUiLCJwcm9maWxlIiwiaWRlbnRpdGllcyIsImdhNGdoX3Bhc3Nwb3J0X3YxIl0sInN1YiI6ImljX2ZiOGMyNGYxMWNkYjQ5MGU4ODE1NzQifQ.B7KUVxofXA_5PBgIN4qEIBl3vhXntgoGKNJ0AE-FzC2X_vLLBSeZ4FpGIjHjFr75XHG0-kPU1wwiz7_wmwsJpwMhybvRyn5f2PjFfzf6HrD3cEkJiz0PnIxkH4Eb3xksOJMhAfZh1ZQlbKXdcc54oDmM3JTyu-YqIQT1J1kBmAqWM3Wy_YX2KtSZHJTQHVxDhFdq9EQosraNJT7wH9lFPPZmNq5YXQVOw7V4lB9UBOFC4sSwnS332xPN_tmzxcobmzEVHpA3UwKI1QfxXQRLxpllavlizN80hJlQgXGIk-GYgFe92JpNbhx2H0ipiUj16e7uKpmwtI3myoXPkQ8wOeVB2ofuLP-FZxojLY4fuUEUA7hg0Gmu6dvV9sz5bVHmxlaIQQl7Pjvz5FlJ7fWWOOTvOLjwMNM546b9Oe_tdovv2cqyQI4f6p6Q0Bk5O7v4-BCd8RWHgWTaA_-GQ0ixn26YUkFUyCilhR4TFJE_0t1Dsyb-wlwc1Op0oTjfilpRlCaOT9AhsMnLWqvtKFhMhIis2_vDCwNSDfNwNXDe-rTCMPqxR42jCfpBOQ8SPSs8YHyLYCGXPTDnsm0zwYg2imM-palKBeJM6RGkQOzBkCA3T4IYuDnxZJekpxGbSluPeruPwc02EjB7Q2qWHw3jnPPSTslwjsw1IieCcxdFmUc"
+
+  "getUserInfoFromJwt" should "get user info for valid jwt" in {
+    val directoryDAO = new MockDirectoryDAO()
+    val googleSubjectId = GoogleSubjectId(genRandom(System.currentTimeMillis()))
+    val uid = genWorkbenchUserId(System.currentTimeMillis())
+    val someIdentityConcentratorId = Some(IdentityConcentratorId("ic_fb8c24f11cdb490e881574"))
+    val email = genNonPetEmail.sample.get
+    directoryDAO.createUser(WorkbenchUser(uid, Some(googleSubjectId), email, someIdentityConcentratorId)).unsafeRunSync()
+    val userInfo = getUserInfoFromJwt(s"Bearer $validJwt", directoryDAO).unsafeRunSync()
+    // jwt was generated a while ago so expiresIn should be negative
+    assert(userInfo.tokenExpiresIn < 0)
+    userInfo.copy(tokenExpiresIn = 0) shouldEqual UserInfo(OAuth2BearerToken(validJwt), uid, email, 0)
+  }
+
+  it should "404 for valid jwt but user does not exist" in {
+    val directoryDAO = new MockDirectoryDAO()
+    val t = intercept[WorkbenchExceptionWithErrorReport] {
+      getUserInfoFromJwt(s"Bearer $validJwt", directoryDAO).unsafeRunSync()
+    }
+    t.errorReport.statusCode shouldBe Some(StatusCodes.NotFound)
+  }
+
+  it should "400 with no exp" in {
```

Comment:
I thought we are throwing 401 code - Unauthorized?

Reply:
indeed, will fix